app/src/main/java/com/lagradost/cloudstream3/extractors/Sendvid.kt

```diff
@@ -0,0 +1,32 @@
+package com.lagradost.cloudstream3.extractors
+
+import com.lagradost.cloudstream3.utils.*
+import com.lagradost.cloudstream3.app
+import com.lagradost.cloudstream3.utils.M3u8Helper.Companion.generateM3u8
+
+class SendvidHttps: Sendvid() {
+    override val mainUrl: String = "https://www.sendvid.com"
+}
+
+open class Sendvid : ExtractorApi() {
+    override var name = "Sendvid"
+    override val mainUrl = "https://sendvid.com"
+    override val requiresReferer = false
+
+    override suspend fun getUrl(url: String, referer: String?): List<ExtractorLink>? {
+        val doc = app.get(url).document
+        val urlString = doc.select("head meta[property=og:video:secure_url]").attr("content")
+        val sources = mutableListOf<ExtractorLink>()
+        if (urlString.contains("m3u8"))  {
+                generateM3u8(
+                    name,
+                    urlString,
+                    mainUrl,
+                    headers = app.get(url).headers.toMap()
```

Comment:
This looks like an exceedingly unnecessary request :sob: 
Please do something like `val response = app.get(url)` and then get the headers and doc separately from that response instead of making the request twice.

Reply:
We dont have that good caching in cs3 :skull: